objects_01.ts

```diff
@@ -0,0 +1,69 @@
+class Warrior {
+    public readonly name: string;
+    public readonly damage: number;
+    public hitPoints: number;
+ 
+    constructor(name: string, damage: number, hitPoints: number) {
+        this.name = name;
+        this.damage = damage;
+        this.hitPoints = hitPoints;
+    }
+    attack(defender: Warrior):void {
+        defender.hitPoints -= this.damage;
+    }
+}
+
+class Battle {
+    private teams: Warrior[][];
+    private messages: Array<string>;
+    constructor() {
+        this.teams = [
+            [new Warrior('Ланселот', 3, 25), new Warrior('Сокол', 3, 25), new Warrior('Храбрый рыцарь', 3, 25), new Warrior('Великий воин', 3, 25)],
+            [new Warrior('Черная рука', 6, 20), new Warrior('Кривой рог', 6, 20), new Warrior('Головорез', 6, 20), new Warrior('Хромой', 6, 20)]
+        ];
+        this.messages= ['откусывает ухо', 'пинает по голове', 'отрубает палец на ноге', 'дергает за волосы', 'выбил зуб'];
+    }
+
+    run() {
+        let attackTeam: Warrior[] = this.teams[0];
+        let defendTeam: Warrior[] = this.teams[1];
```

Comment:
можно ставить тайпинги, но не обязательно, т.к. п правой части вывод сработает

```suggestion
        let attackTeam = this.teams[0];
        let defendTeam = this.teams[1];
```